wrappers/src/object_cs.cpp

```diff
@@ -219,175 +219,182 @@ extern "C" {
+            verify_can_set(object);
             
-            const size_t column_ndx = get_column_index(object_ptr, property_ndx);
-            object_ptr->row().set_link(column_ndx, target_object_ptr->row().get_index());
+            const size_t column_ndx = get_column_index(object, property_ndx);
+            object.row().set_link(column_ndx, target_object.row().get_index());
         });
     }
     
-    REALM_EXPORT void object_clear_link(const Object* object_ptr, size_t property_ndx, NativeException::Marshallable& ex)
+    REALM_EXPORT void object_clear_link(const Object& object, size_t property_ndx, NativeException::Marshallable& ex)
     {
         return handle_errors(ex, [&]() {
-            verify_can_set(object_ptr);
+            verify_can_set(object);
             
-            const size_t column_ndx = get_column_index(object_ptr, property_ndx);
-            object_ptr->row().nullify_link(column_ndx);
+            const size_t column_ndx = get_column_index(object, property_ndx);
+            object.row().nullify_link(column_ndx);
         });
     }
     
-    REALM_EXPORT void object_set_null(const Object* object_ptr, size_t property_ndx, NativeException::Marshallable& ex)
+    REALM_EXPORT void object_set_null(const Object& object, size_t property_ndx, NativeException::Marshallable& ex)
     {
         return handle_errors(ex, [&]() {
-            verify_can_set(object_ptr);
+            verify_can_set(object);
             
-            const size_t column_ndx = get_column_index(object_ptr, property_ndx);
-            if (!object_ptr->row().get_table()->is_nullable(column_ndx))
+            const size_t column_ndx = get_column_index(object, property_ndx);
+            if (!object.row().get_table()->is_nullable(column_ndx))
                 throw std::invalid_argument("Column is not nullable");
             
-            object_ptr->row().set_null(column_ndx);
+            object.row().set_null(column_ndx);
         });
     }
     
-    REALM_EXPORT void object_set_bool(const Object* object_ptr, size_t property_ndx, size_t value, NativeException::Marshallable& ex)
+    REALM_EXPORT void object_set_bool(const Object& object, size_t property_ndx, size_t value, NativeException::Marshallable& ex)
     {
         return handle_errors(ex, [&]() {
-            verify_can_set(object_ptr);
+            verify_can_set(object);
             
-            const size_t column_ndx = get_column_index(object_ptr, property_ndx);
-            object_ptr->row().set_bool(column_ndx, size_t_to_bool(value));
+            const size_t column_ndx = get_column_index(object, property_ndx);
+            object.row().set_bool(column_ndx, size_t_to_bool(value));
         });
     }
     
-    REALM_EXPORT void object_set_int64(const Object* object_ptr, size_t property_ndx, int64_t value, NativeException::Marshallable& ex)
+    REALM_EXPORT void object_set_int64(const Object& object, size_t property_ndx, int64_t value, NativeException::Marshallable& ex)
```

Comment:
I also don't understand how these compile with `const` objects, as `set_int` obviously mutates. In any case, it feels wrong to me to declare a parameter that is being manipulated `const`.